### ISSUE TITLE
RBF Custom Layer in Tensorflow can not learn the centers (mu) parameters

### ISSUE THREAD

**Issue by miladtoutounchian** [tensorflow/tensorflow#50584]
Using Google Colab with TF version 2.5

**Comment by miladtoutounchian** [comment]
[Bug.zip](https://github.com/tensorflow/tensorflow/files/6755781/Bug.zip)

**Comment by miladtoutounchian** [comment]
from tensorflow.keras.layers import Layer
from tensorflow.keras import backend as K

class RBFLayer(Layer):
    def __init__(self, units, gamma, **kwargs):
        super(RBFLayer, self).__init__(**kwargs)
        self.units = units
        self.gamma = K.cast_to_floatx(gamma)

    def build(self, input_shape):
        self.mu = self.add_weight(name='mu',
                                  shape=(int(input_shape[1]), self.units),
                                  initializer=tf.random_normal_initializer(),
                                  trainable=True)
        # self.gamma = self.add_weight(name='gamma',
        #                           shape=(int(input_shape[1]),),
        #                           initializer=tf.constant_initializer(1),
        #                           trainable=True)
        super(RBFLayer, self).build(input_shape)

    def call(self, inputs):
        diff = K.expand_dims(inputs) - self.mu
        l2 = K.sum(K.pow(diff,2), axis=1)
        res = K.exp(-0.5*self.gamma * l2)
        return res

    def compute_output_shape(self, input_shape):
        return (input_shape[0], self.units)

**Comment by saikumarchalla** [comment]
Was able to reproduce the issue in TF 2.5. Please find the gist [here](https://colab.research.google.com/drive/1REr597ifuYcSsM9Bm27C6OuyGYIgUzTg#scrollTo=tg2DqpDg9PLQ). Thanks!

**Comment by Saduf2019** [comment]
@miladtoutounchian 
Please post this issue on keras-team/[keras repo](https://github.com/keras-team/keras/issues).
To know more refer to:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
